src/serac/physics/state/state_manager.hpp

```diff
@@ -95,19 +95,119 @@ class StateManager {
+                                std::string(geom_name)));
+          axom::sidre::Group* geom_group = qdatas_group->getGroup(std::string(geom_name));
+
+          // Verify size correctness
+          auto verify_size = [](axom::sidre::Group* group, int value, const std::string& view_name,
+                                const std::string& err_msg) {
+            SLIC_ERROR_IF(
+                !group->hasView(view_name),
+                axom::fmt::format("Loaded Sidre Datastore does not have value '{}' for Quadrature Data.", view_name));
+            auto prev_value = group->getView(view_name)->getData<axom::IndexType>();
+            SLIC_ERROR_IF(value != prev_value, axom::fmt::format(err_msg, value, prev_value));
+          };
+          verify_size(geom_group, num_states, "num_states",
+                      "Current number of Quadrature Data States '{}' does not match value in restart '{}'.");
+          verify_size(geom_group, state_size, "state_size",
+                      "Current size of Quadrature Data State '{}' does not match value in restart '{}'.");
+          verify_size(geom_group, total_size, "total_size",
+                      "Current total size of Quadrature Data States '{}' does not match value in restart '{}'.");
+
+          // Tell Sidre where the external array is
+          SLIC_ERROR_ROOT_IF(!geom_group->hasView("states"),
+                             "Loaded Quadrature Data geometry Sidre view did not have 'states'");
+          axom::sidre::View* states_view = geom_group->getView("states");
+          states_view->setExternalDataPtr(states.data());
+        }
+      }
+    }
+
+    if (is_restart_) {
+      // NOTE: This call will reload all external buffers from file stored in the DataStore
+      // TODO: This should be changed to load only the current material quadrature data after
+      // MFEMSidreDatacollection::LoadExternalData and SPIO is enhanced to allow loading the
+      // external data piecemeal
+      datacoll.LoadExternalData();
+    }
+  }
+
   /**
    * @brief Create a shared ptr to a quadrature data buffer for the given material type
    *
    * @tparam T the type to be created at each quadrature point
+   * @param mesh_tag The tag for the stored mesh used to locate the datacollection
    * @param domain The spatial domain over which to allocate the quadrature data
    * @param order The order of the discretization of the primal fields
    * @param dim The spatial dimension of the mesh
    * @param initial_state the value to be broadcast to each quadrature point
    * @return shared pointer to quadrature data buffer
    */
   template <typename T>
-  static std::shared_ptr<QuadratureData<T>> newQuadratureDataBuffer(const Domain& domain, int order, int dim,
-                                                                    T initial_state)
+  static std::shared_ptr<QuadratureData<T>> newQuadratureDataBuffer(const std::string& mesh_tag, const Domain& domain,
```

Comment:
The mesh holds domains now. We will want to eventually pass the string identifying the domain instead of passing the domain object, but I don't think we're ready to do that yet. We need to flesh out our mesh class more fully. Nothing to do here, just a note.